libs/base/Data/SnocList.idr

```diff
@@ -62,9 +62,26 @@ Show a => Show (SnocList a) where
       show' acc (xs :< x) = show' (show x :: acc) xs
 
 public export
+mapImpl : (a -> b) -> SnocList a -> SnocList b
+mapImpl f Lin = Lin
+mapImpl f (sx :< x) = (mapImpl f sx) :< (f x)
+
+-- Utility for implementing `mapTR`
+mapTR' : List b -> (a -> b) -> SnocList a -> SnocList b
```

Comment:
Should this be called `mapAppend` for uniformity?

Reply:
> Should this be called `mapAppend` for uniformity?

`Data.SnocList.mapAppend` already exists. But I'm not very good at choosing useful names so I'm happy to change this to something else.

Reply:
No one has piped up with a better name and this is a net gain so let's merge it now, we can always rename later on.